keps/sig-storage/3751-volume-attributes-class/README.md

```diff
@@ -685,11 +716,78 @@ spec:
+  - Infeasible Errors (e.g., `InvalidArgument`): This is a subset of final errors indicating the request itself is invalid and is not likely to succeed when retried.
+    We will retry at lower frequency.
+    If the `Spec.VolumeAttributesClassName` is set to nil, we will not retry the request.
+
+For details, please refer to the flow chart below.
+
+##### Implementation
+
+Flow diagram describes how external-resizer modifies the volume:
+![ModifyVolume Flow Diagram](./VolumeAttributesClass-Modify-Flow.jpg)
+where
+* `spec`: `pvc.spec.volumeAttributesClassName`
+* `cur`: `pvc.status.currentVolumeAttributesClassName`
+* `target`: `pvc.status.modifyVolumeState.targetVolumeAttributesClassName`
+* `status`: `pvc.status.modifyVolumeState.status`. resetting status to nil also implies resetting `target` to nil.
+
+To further verify the correctness, this tables lists all possible states and the corresponding actions:
+
+| \# | spec | cur | status | target | uncertain | action | OK | final | infeasible | non-final |
+| -: | --- | --- | ------ | --- | ----- | -: | -: | -: | -: | -: |
+|  1 | nil | nil | nil    | N/A | false |  1 |    |    |    |    |
+|  2 | nil | nil | InProg | A   | \*    |  A |  4 |  2 |  3 |  2 |
+|  3 | nil | A   | Infeas | A   | \*    |  3 |    |    |    |    |
+|  4 | nil | A   | nil    | N/A | false |  4 |    |    |    |    |
+|  5 | A   | nil | nil    | N/A | false |  6 |    |    |    |    |
+|  6 | A   | nil | InProg | A   | false |  A |  9 |  6 |  8 |  7 |
+|  7 | A   | nil | InProg | A   | true  |  A |  9 |  6 |  8 |  7 |
+|  8 | A   | nil | Infeas | A   | false |  6 |    |    |    |    |
+|  9 | A   | A   | nil    | N/A | false |  9 |    |    |    |    |
+| 10 | B   | nil | InProg | A   | false |  6 |    |    |    |    |
+| 11 | B   | nil | InProg | A   | true  |  A | 13 | 10 | 12 | 11 |
+| 12 | B   | nil | Infeas | A   | false |  6 |    |    |    |    |
+| 13 | B   | A   | nil    | N/A | false | 14 |    |    |    |    |
+| 14 | B   | A   | InProg | B   | false |  B |  9 | 14 | 16 | 15 |
+| 15 | B   | A   | InProg | B   | true  |  B |  9 | 14 | 16 | 15 |
+| 16 | B   | A   | Infeas | B   | false | 14 |    |    |    |    |
+| 17 | C   | A   | InProg | B   | false | 14 |    |    |    |    |
+| 18 | C   | A   | InProg | B   | true  |  B | 13 | 17 | 19 | 18 |
+| 19 | C   | A   | Infeas | B   | false | 14 |    |    |    |    |
+| 20 | A   | A   | InProg | B   | false | 23 |    |    |    |    |
+| 21 | A   | A   | InProg | B   | true  |  B | 13 | 20 | 22 | 21 |
+| 22 | A   | A   | Infeas | B   | false | 23 |    |    |    |    |
+| 23 | A   | A   | InProg | A   | false |  A |  9 | 23 | 25 | 24 |
+| 24 | A   | A   | InProg | A   | true  |  A |  9 | 23 | 25 | 24 |
+| 25 | A   | A   | Infeas | A   | false | 23 |    |    |    |    |
+| 26 | B   | A   | InProg | A   | false | 14 |    |    |    |    |
+| 27 | B   | A   | InProg | A   | true  |  A | 13 | 26 | 28 | 27 |
+| 28 | B   | A   | Infeas | A   | false | 14 |    |    |    |    |
```

Comment:
It is not clear tbh, what those action mean.  What does Action-6 mean?

Reply:
Added an example and some more descriptions to this table. Hopes this will help.